zebra-chain/src/primitives/zcash_primitives.rs

```diff
@@ -290,8 +290,15 @@ pub(crate) fn transparent_output_address(
     output: &transparent::Output,
     network: Network,
 ) -> Option<transparent::Address> {
-    let script = zcash_primitives::legacy::Script::from(&output.lock_script);
-    let alt_addr = script.address();
+    // Create a `TxOut` primitive from a zebra `Output`.
+    let tx_out = zcash_primitives::transaction::components::TxOut {
+        // We just set the value to zero as we are only interested in the
+        // address derived from the output lock script.
+        value: zcash_primitives::transaction::components::Amount::zero(),
+        script_pubkey: zcash_primitives::legacy::Script::from(&output.lock_script),
+    };
+    let alt_addr = tx_out.recipient_address();
```

Comment:
This code feels like a bit of a workaround, which could break if `librustzcash` changes.

If the address is only valid in the context of an output, then we should convert the whole output before getting the address. (It doesn't seem to make any difference now, but the `zcashd` developers suggested using the whole output.)

I made this change in PR #5507, what do you think?

Reply:
Yea, that looks a bit better, I merged suggestion to here.